### PR TITLE
Integrate 'endpointId' returned by the backend service

### PR DESCRIPTION
**Pull request by paweldomas**

### CODE REVIEW

File: spot-client/src/common/backend/SpotBackendService.js [hunk 1]
```diff
@@ -217,13 +224,30 @@ export class SpotBackendService extends Emitter {
...
+                = registerDevice(`${this.pairingServiceUrl}`, pairingCode, storedEndpointId)
+                    .then(registration => {
+                        const { endpointId, ...rest } = registration;
+
+                        if (storedEndpointId !== endpointId) {
+                            logger.log('Storing new endpoint ID', {
+                                endpointId,
+                                endpointIdPersistenceKey: this.endpointIdPersistenceKey
+                            });
+                            persistence.set(this.endpointIdPersistenceKey, endpointId);
```
Comment (virtuacoplenny) [inline spot-client/src/common/backend/SpotBackendService.js]: what if we set and log each time no matter what? Could that be safer? Worse?

File: spot-client/src/common/backend/SpotBackendService.js [hunk 2]
```diff
@@ -217,13 +224,30 @@ export class SpotBackendService extends Emitter {
         }
 
         if (!registerDevicePromise) {
+            const storedEndpointId = persistence.get(this.endpointIdPersistenceKey);
```
Comment (virtuacoplenny) [inline spot-client/src/common/backend/SpotBackendService.js]: Are there any cases where the endpointIdPersistenceKey needs to be cleared? The point of it is not to be cleared?

Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.js]: It seems that this one should not be cleared.

Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.js]: At least at this point I'm not aware of any cases when it should be cleared.

File: spot-client/src/common/backend/SpotBackendService.js (hunk 1 above)
Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.js]: If register request is going to happen, then another message will be logged above which should contain the endpoint ID already(if it exists).
I'm not sure if we should store every time just for safety. Eventually it's been retrieved from storage already, so it it there?

**Pull request merged**